exec/selected/selected.go

```diff
@@ -55,6 +55,43 @@ type SchemaField struct {
 	FixedResult reflect.Value
 }
 
+func (sf *SchemaField) ToSelectedField() *types.SelectedField {
+	if sf == nil {
+		return nil
+	}
+	return &types.SelectedField{
+		Name:       sf.Name,
+		Alias:      sf.Alias,
+		Fields:     selsToSelectedFields(sf.Sels),
+		Args:       sf.Args,
+		Directives: sf.Directives,
+	}
+}
+
+func selsToSelectedFields(sels []Selection) (fs []*types.SelectedField) {
+	for _, f := range sels {
+		// Selection is SchemaField or TypeAssertion or TypenameField
+		switch v := f.(type) {
+		case *SchemaField:
+			fs = append(fs, v.ToSelectedField())
+		case *TypeAssertion:
+			// TypeAssertion is a selection dependent on the type of a union field
+			var assertedTypeName string
+			if v, ok := v.TypeAssertion.TypeExec.(*resolvable.Object); ok {
+				assertedTypeName = v.Name
+			}
+			for _, f := range selsToSelectedFields(v.Sels) {
```

Comment:
please dont shadow `f` its too confusing

unfortunately this still does not support graphql interfaces "properly", as in you'll need to be aware of the relationships between the types in the resolver code (and the data is present in _this_ layer, rather than there)

consider if the fact that it's an interface actually matters - and use the context present in `TypeExec` to provide info to allow the caller to map fields to possible "union" types, rather than to interfaces